modules/EmailMan/EmailMan.php

```diff
@@ -1003,6 +1003,179 @@ public function mark_deleted($id)
         $this->db->query("DELETE FROM {$this->table_name} WHERE id=" . intval($id));
     }
 
+    /**
+     *
+     * @global array $sugar_config
+     * @param string $module
+     * @param string $uid
+     * @return boolean
+     */
+    public function addOptInEmailToEmailQueue($module, $uid) {
+
+        global $sugar_config;
+
+        $confirmOptInEnabled = isset($sugar_config['email_enable_confirm_opt_in']) && $sugar_config['email_enable_confirm_opt_in'];
```

Comment:
`email_enable_confirm_opt_in` are changed to string-enum but still using as a boolean, which is not a problem if we want to add Opt In Emails to Email Queue each case when this settings are not disabled (as `''` - empty string). If not it should be changed to check the correct enum value.
Note: this check are many places, may better practice if we outsource this into a commonly available function to avoid any code duplication.